### DISCUSSION TITLE
allow templates/string injection in request transformation

### DISCUSSION THREAD

**Discussion by ahmadnassri** [Kong/kong#7233] [Ideas and feature requests]
###### USE CASE

when doing consumer level transformations, it can get complex real fast, e.g.

``` js
{
  "consumer_id": "consumer-id-1",
  "api_id": "api-id",
  "name": "request-transformer",
  "config": {
    "add": {
      "querystring": {
        "consumer_id": "consumer-id-1" // manually plot the consumer id into a query string
      },
      "body": {},
      "headers": {}
    }
  }
}, {
  "consumer_id": "consumer-id-2",
  "api_id": "api-id",
  "name": "request-transformer",
  "config": {
    "add": {
      "querystring": {
        "consumer_id": "consumer-id-2" // manually plot the consumer id into a query string
      },
      "body": {},
      "headers": {}
    }
  }
}
```

a better alternative would be: 

``` json
{
  "api_id": "api-id",
  "name": "request-transformer",
  "config": {
    "add": {
      "querystring": {
        "remote_api_key: {{ CONSUMER_ID }}"
      },
      "body": {},
      "headers": {}
    }
  }
}
```

this can become very useful when coupled with https://github.com/Mashape/kong/issues/550

other use cases include injecting additional properties from the execution logic, e.g. cluster info, server address, ports, etc ...

**Comment by jmdacruz** [comment]
I can also think about more complex transformations on the `request-transformer` plugin, such as duplicating headers with different names, transforming values of headers into query string parameters and viceversa, etc.

**Comment by JnMik** [comment]
` + 1`
